spaceMissionManager-service/src/main/java/cz/muni/fi/services/impl/MissionServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package cz.muni.fi.services.impl;
+
+import cz.muni.fi.dao.MissionDao;
+import cz.muni.fi.entity.Mission;
+import cz.muni.fi.services.MissionService;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation of the {@link MissionService}. This class is part of the
+ * service module of the application that provides the implementation of the
+ * business logic (main logic of the application).
+ *
+ * @author Vojtech Bruza
+ */
+@Service
+public class MissionServiceImpl implements MissionService {
+    @Inject
+    MissionDao missionDao;
+
+    @Override
+    public void archive(Long missionId, LocalDate endDate){
+        if(endDate.isAfter(LocalDate.now())){
+            throw new IllegalArgumentException("Mission end date must be in the past");
+        }
+        Mission mission = missionDao.findMissionById(missionId);
```

Comment:
findMissionById will return null if missionId is not in DB, possible NullPointerException